Sources/PackageModel/Manifest/PackageConditionDescription.swift

```diff
@@ -9,7 +9,7 @@
 */
 
 /// Represents a manifest condition.
-public struct PackageConditionDescription: Codable, Equatable {
+public struct PackageConditionDescription: Codable, Equatable, Hashable {
```

Comment:
Do you think it is worth also warning of dependencies that can be combined?

For example warning about this:

```swift
.target(name: "a", condition: .when(platforms: [.macOS])),
.target(name: "a", condition: .when(platforms: [.linux]))
```

And suggesting this:

```swift
.target(name: "a", condition: .when(platforms: [.macOS, .linux]))
```

Reply:
That seems like a stylistic choice to me. At least in Swift itself, this is not something that is enforced, e.g.:
```swift
switch foo {
    case .a:
        print("not c")
    case .b:
        print("not c")
    case .c:
        print("c!")
}
```
vs 
```swift
switch foo {
    case .a, .b:
        print("not c")
    case .c:
        print("c!")
}
```
There is no warning to combine the first two cases.

But maybe the package manager has different conventions?

Reply:
I have no strong opinion either way. It just came up while I was thinking through the implications of comparing by name, and I wondered what your thoughts were.